starport_template/lib/stores/wallets_store.dart

```diff
@@ -268,4 +281,18 @@ class WalletsStore {
       logError(ex, stack);
     }
   }
+
+  Future<void> getTransactionHistory() async {
+    transactionsList.clear();
+    isTransactionHistoryLoading = true;
+    try {
+      transactionsList.addAll(
```

Comment:
if it fails, you'll end up with cleared list and no transactions. you should clear the list just before adding new elements to it